backend/controllers/session.js

```diff
@@ -166,6 +166,60 @@ module.exports.loginStudent = async (req, res) => {
     });
 };
 
+module.exports.resetPassword = async (req, res) => {
+    if (!req.user) {
+        return res.status(403).json({
```

Comment:
We don't need this `if` block. Passport will abort the request before it ever gets to the controller's code.